File: android/walletsdk/src/main/java/com/coinbase/android/nativesdk/task/Task.kt
package com.coinbase.android.nativesdk.task

import com.coinbase.android.nativesdk.message.request.UnencryptedRequestMessage
import com.coinbase.android.nativesdk.message.response.ResponseHandler
import java.util.Date

internal class Task(
    val request: UnencryptedRequestMessage,
    val handler: ResponseHandler,
    val timestamp: Date
)

File: ios/CoinbaseWalletSDK/Task/Task.swift
//
//  Task.swift
//  MobileWalletProtocol
//
//  Created by Jungho Bang on 6/14/22.
//

import Foundation

@available(iOS 13.0, *)
struct Task {
    let request: RequestMessage
    let host: URL
    let handler: ResponseHandler
    let timestamp: Date
}
